lxd/firewall/drivers/drivers_nftables.go

```diff
@@ -405,27 +405,38 @@ func (d Nftables) InstanceSetupBridgeFilter(projectName string, instanceName str
 	}
 
 	// Filter unwanted ethernet frames when using IP filtering.
-	if IPv4 != nil || IPv6 != nil {
+	if len(IPv4Nets)+len(IPv6Nets) > 0 {
 		tplFields["filterUnwantedFrames"] = true
 	}
 
-	if IPv4 != nil {
-		if IPv4.String() == FilterIPv4All {
-			tplFields["ipv4FilterAll"] = true
-		} else {
-			tplFields["ipv4Addr"] = IPv4.String()
-		}
+	var ipv4Nets []string
```

Comment:
Can you move this line to just above the for loop that populates it and change it to pre-allocate the slice, e.g.

```go
ipv4Nets := make([]string, 0, len(IPv4Nets))
```